hydra/redisgraph_demo1.py

```diff
@@ -0,0 +1,258 @@
+            redis_graph.add_edge(edge)
+    if post_endpoint == 0 and put_endpoint == 0:
+        flag = 0
+        for obj in apidoc["supportedClass"]:
+            if obj["@id"] == returns:
+                flag = 1
+                collection = obj
+        if flag == 1:
+            for obj in collection["supportedOperation"]:
+                if obj["method"] == "POST" or obj["method"] == "PUT":
+                    return_class = obj["returns"]
+                    class_node = Node(
+                        label="id",
+                        alias="classes",
+                        properties=node.properties)
+                    redis_graph.add_node(class_node)
+                    break
+                if obj["method"] == "POST":
+                    post_collection = 1
+                if obj["method"] == "PUT":
+                    put_collection = 1
+
+    """ connect the collection to the classes by edge"""
+
+    edge = Edge(collection_node, "has_classes", class_node)
+    redis_graph.add_edge(edge)
+    if (post_endpoint == 1 or
+            put_endpoint == 1 or
+            post_collection == 1 or
+            put_collection == 1):
+        flag = 0
+        for obj in apidoc["supportedclass"]:
+            if obj["@id"] == return_class:
+                flag = 1
+                classes = obj
+        if flag == 1:
+            for obj in classes["supportedOperation"]:
+                if obj["method"] == "PUT":
+                    put_classes = 1
+                if obj["method"] == "POST":
+                    post_classes = 1
+                if obj["method"] == "DELETE":
+                    delete_classes = 1
+                if obj["method"] == "GET":
+                    get_classes = 1
+
+    """ members are leave nodes which is connect with the classes
+        every elementary operation should be perform on these.
+        like put,post,delete these operation can be done by adding and
+        deleting the nodes and update all the above nodes properties"""
+
+    for member in new_file["members"]:
+        endpoint_member = {}
+        endpoint_member["@id"] = str(member["@id"])
+        response = urllib.urlopen(base_url + member["@id"])
+        new_dict = json.loads(response.read())
+        endpoint_member["@type"] = str(new_dict["@type"])
+        endpoint_member["name"] = str(new_dict["name"])
+        endpoint_member["description"] = str(new_dict["description"])
+        endpoint_member["start_date"] = str(new_dict["start_date"])
+        endpoint_member["end_date"] = str(new_dict["end_date"])
+        endpoint_member_alias = str(get_member_alias(member["@id"]))
+        print endpoint_member_alias
+        node1 = Node(
+            label="endpoint_member",
+            alias=str(endpoint_member_alias),
+            properties=endpoint_member)
+        redis_graph.add_node(node1)
+        edge = Edge(class_node, "class_get_member", node1)
+        redis_graph.add_edge(edge)
```

Comment:
Please try to make your code modular and reusable by breaking it into smaller functions. The practice you're following is a really bad one.

Reply:
Here are some practices you should try to follow https://stevenloria.com/python-best-practice/

Reply:
The above errors are occurring due to the Redis set up, I 'll provide a setup file for running the same file with Redis. And yeah I am working on it, I'll break it in reusable code functions. Actually, I  want to show that this graph can work finely? can I move forward with this graph after some required change in it?

Reply:
I need to run your code in order to understand what it does